src/main/java/frc/robot/services/LedsService.java

```diff
@@ -0,0 +1,86 @@
+package frc.robot.services;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import frc.robot.RobotMap;
+
+public class LedsService {
+
+    private static final int NUMBER_OF_LEDS = 60;
+
+    private LedsService(AddressableLED led, AddressableLEDBuffer ledBuffer, VisionService vision) {
+        this.led = led;
+        this.ledBuffer = ledBuffer;
+        this.vision = vision;
+        led.setLength(ledBuffer.getLength());
+        led.start();
+    }
+
+    private static LedsService instance;
+
+    private final VisionService vision;
+
+    private final AddressableLED led;
+    private final AddressableLEDBuffer ledBuffer;
+
+    public static LedsService getInstance() {
+        if (instance == null) {
+            instance = new LedsService(new AddressableLED(RobotMap.PWM.LED_PORT),
+                    new AddressableLEDBuffer(NUMBER_OF_LEDS), VisionService.getInstance());
+        }
+        return instance;
+    }
+
+    public enum Mode {
+
+        EMPTY_GRIPPER(254, 0, 0), ALLIGNED(204, 0, 254), HAS_GAME_PIECE(0, 0, 254),
+        HAS_GAME_PIECE_AND_ALLIGNED(0, 254, 0);
+
+        public final int red;
+        public final int green;
+        public final int blue;
+
+        Mode(int red, int green, int blue) {
+            this.red = red;
+            this.green = green;
+            this.blue = blue;
+        }
+    }
+
+    public void periodic() {
+        boolean hasTarget = vision.limelightHasTarget();
```

Comment:
we want to know if it's aligned, not whether it has target or not
need to add methods in VisionService:
limelightCentered (check if limelight has target, if it has return whether tx is between -1 and 1 (for tolerance))
photonVisionCentered (same thing but with photonvision, check yaw instead of tx)